### ISSUE TITLE
Consider renaming `fail_map` to `error_map`

### ISSUE THREAD

**Issue by mre** [lycheeverse/lychee#1446]
We no longer use the term `failure` in our system; we now refer to it as `error`. Therefore, it makes sense to rename `fail_map` to `error_map` for consistency. Similar to #1443, this will introduce a breaking change, but since we haven't reached a stable version yet, I want to address these naming inconsistencies beforehand.

**Comment by mre** [comment]
Related discussion, which could be tackled at the same time: https://github.com/lycheeverse/lychee/discussions/1013

**Comment by thomas-zahner** [comment]
I totally agree with that change :+1: 

As for #1013 this was actually already implemented in https://github.com/lycheeverse/lychee/commit/250f7a8f0af00e927ebed6a0b706d5833e480779. Although, I think that the `text` field is a bit verbose, maybe we could reduce to just the HTTP status text.

The current JSON output looks like:

```json
  "success_map": {
    "http://www.example.com/": [
      {
        "url": "https://www.iana.org/domains/example",
        "status": {
          "text": "OK (200 OK)",
          "code": 200
        }
      }
    ]
  },

```

**Comment by mre** [comment]
Yup, sounds like a good idea!

**Comment by mre** [comment]
I took a stab at it in #1560.
The status text has already been fixed with #1553. It looks a little cleaner now:


```json
"success_map": {
  "http://www.example.com/": [
    {
      "url": "https://www.iana.org/domains/example",
      "status": {
        "text": "200 OK",
        "code": 200
      }
    }
  ]
}
```
